src/ProgressOnderwijsUtils/Collections/MaybeExtensions.cs

```diff
@@ -278,28 +263,55 @@ public static void IfError<TOk>(this Maybe<TOk, Unit> state, Action ifError)
         [Pure]
         public static bool ContainsError<TOk, TError>(this Maybe<TOk, TError> state, Predicate<TError> predicate)
         {
-            var (isOk, _, errValue) = state.ExtractionHelper();
-            return !isOk && predicate(errValue);
+            var isOk = state.TryGet(out _, out var error);
+            return !isOk && predicate(error);
         }
 
         [Pure]
         public static Maybe<TOk[], TError[]> WhenAllOk<TOk, TError>([NotNull] this IEnumerable<Maybe<TOk, TError>> maybes)
+        {
+            var (okValues, errValues) = maybes.Partition();
+            return errValues.Any() ? (Maybe<TOk[], TError[]>)Maybe.Error(errValues) : Maybe.Ok(okValues).AsMaybeWithoutError<TError[]>();
+        }
+
+        [NotNull]
+        [Pure]
+        public static IEnumerable<TOk> WhereOk<TOk, TError>([NotNull] this IEnumerable<Maybe<TOk, TError>> maybes)
+        {
+            foreach (var state in maybes) {
+                var isOk = state.TryGet(out var okValue, out _);
+                if (isOk) {
+                    yield return okValue;
+                }
+            }
+        }
+
+        [NotNull]
+        [Pure]
+        public static IEnumerable<TError> WhereError<TOk, TError>([NotNull] this IEnumerable<Maybe<TOk, TError>> maybes)
+        {
+            foreach (var state in maybes) {
+                var isOk = state.TryGet(out _, out var error);
+                if (!isOk) {
+                    yield return error;
+                }
+            }
+        }
+
+        [Pure]
+        public static (TOk[] okValues, TError[] errorValues) Partition<TOk, TError>([NotNull] this IEnumerable<Maybe<TOk, TError>> maybes)

```

Comment:
Added: way to partition a list of maybes into the ok values, and the error values.